Fixes of i18n on vtex-io-documentation-best-practices-for-optmizing-performance.md (#525)
vtexdocs/dev-portal-content: docs/vtex-io/Perks/best-practices/vtex-io-documentation-best-practices-for-optimizing-performance.md

> ℹ️ More details on how to implement this optimization can be found in the documentation on [Enabling OrderForm optimization](https://developers.vtex.com/docs/guides/vtex-io-documentation-enabling-order-form-optimization/).

### Deactivating the VTEX IO native service worker

The VTEX IO platform provides a native service worker to every store that uses Store Framework. However, since a website can only run one service worker at a time, you may face conflicts when working with a third-party service that provides its own service worker.

You can deactivate the native service worker provided by VTEX IO to successfully use a third-party solution in your store.

> ℹ️ For more information, read our documentation on [Deactivating the VTEX IO native service worker](https://developers.vtex.com/docs/guides/vtex-io-documentation-deactivating-the-vtex-io-native-service-worker/).

### Optimizing critical CSS

> The content which a user first sees when opening a web page is known as *above-the-fold* content. This part of the page can also be called *critical* because it needs to be loaded quickly to provide a better user experience.

Reviewer: ```suggestion
The content which a user first sees when opening a web page is known as above-the-fold content. This part of the page can also be called critical because it needs to be loaded quickly to provide a better user experience.
```

![critical](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/images/vtex-io-documentation-best-practices-for-optimizing-performance-1.png)

In other words, the critical part of the page must be rendered correctly and promptly before anything else.

> However, by default, the browser only renders a web page after it has finished loading, parsing, and executing all related CSS files. As a result, the time taken by a browser to render the critical part of the page can significantly increase for large CSS codes.

Reviewer: ```suggestion
However, by default, the browser starts the rendering process of a web page once it has completed loading, parsing, and executing all related CSS files. Consequently, when dealing with large CSS code, the browser may experience a notable increase in the time required to render the critical part of the page.
```

> Fortunately, VTEX IO can enable critical CSS optimization on home pages. This feature provides the browser a way to find the *minimum blocks of CSS code* first needed to display the critical content of the page, while the remainder of the CSS code is loaded asynchronously.

Reviewer: ```suggestion
To address this problem, you can enable critical CSS optimization on home pages. This feature allows the browser to identify the essential CSS code blocks required to display the critical part of the page while loading the remaining CSS code asynchronously.
```

> ⚠️ Keep in mind that by enabling this option, you might notice some style inconsistencies in the store layout. Test it with caution before enabling it on a production workspace. Also, if you notice any side-effects on the store website, please [open a ticket](https://help-tickets.vtex.com/smartlink/sso/login/zendesk) and let us know!

### Improving download speed with CSS concatenation

To correctly display a web page layout, the browser needs to make multiple HTTP requests to download each CSS file. However, numerous requests might increase the time to render a single web page.

VTEX IO can avoid various download requests by concatenating multiple CSS files in one.

![concatenation](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/images/vtex-io-documentation-best-practices-for-optimizing-performance-2.png)

Note that by combining CSS in a single file, the browser makes fewer requests and loads web pages faster, improving the performance of your ecommerce website.

### Lazy loading page metadata

> Before displaying a web page, the VTEX IO service app for server-side rendering, or simply Render Runtime, interprets the page metadata script. You can see an example of this script below.

Reviewer: ```suggestion
Before displaying a web page, the VTEX IO service app for server-side rendering (Render Runtime) interprets the page metadata script. You can see an example of this script below.
```

```ts
<script>
        __RUNTIME__ = {"account":"vtexstore","amp":false,"bindingChanged":false,"binding":{"id":"aacb04t3-a8fa-4bab-b5bd-2d654d20dcd8","canonicalBaseAddress":"vtexstore.vtex.com"},"culture":{"availableLocales":[],"country":"USA","currency":"USD","language":"en","locale":"en-US","customCurrencyDecimalDigits":null,"customCurrencySymbol":"$"},"production":true,"query":{},"settings":{....
```

Note that this script holds a lot of important data about a web page, and it can be long, which can be a big task for the browser.

> Therefore, to prevent the total blocking time of the store website from being compromised, this script will be broken into smaller ones when lazy runtime is enabled.

Reviewer: ```suggestion
Therefore, when lazy runtime is enabled, this script is broken into smaller ones in order to prevent the total blocking time of the store website from being compromised.
```

### Improving search results

If the search results page is flooded with product information, performance problems can arise when loading or rendering its data.

Fortunately, this can be changed with two props, namely `skusFilter` and `simulationBehavior`.

These props control the SKUs returned for each product in the search query and define whether the displayed search data will be up-to-date (`skusFilter`) or fetched using the cache (`simulationBehavior`).

We recommend that you only allow the first available SKU to be returned for each product (using the `FIRST_AVAILABLE` value in the `skusFilter` prop) and use cache to display the search data (using the `skip` value in the `simulationBehavior` prop).

> ℹ️ Learn more about how to properly configure this in the [Search Results app documentation](https://developers.vtex.com/docs/guides/vtex-search-result).

### Lazy loading of images and product information in a slider

We recommend you build your theme shelf and carousel blocks using Slider Layout.

> Slider Layout natively lazily loads the rendered images or product information. That means that the information in a shelf or carousel built with Slider will only be loaded when components receive user interaction, decreasing the website page load time.

Reviewer: ```suggestion
Slider Layout natively lazy loads the rendered images or product information. That means that the information in a shelf or carousel built with Slider will only be loaded when components receive user interaction, decreasing the website page load time.
```
